packages/flutter/lib/src/rendering/object.dart

```diff
@@ -4751,726 +4501,1163 @@ mixin RelayoutWhenSystemFontsChangeMixin on RenderObject {
+          semanticsClipRect: cachedSemanticsNode!.parentSemanticsClipRect,
+        ),
+        usedSemanticsIds: <int>{},
         elevationAdjustment: 0.0,
-        result: children,
-        siblingNodes: childSiblingNodes,
       );
-      siblingNodes.addAll(childSiblingNodes);
     }
+  }
 
-    if (_config.isSemanticBoundary) {
-      owner.assembleSemanticsNode(node, _config, children);
-    } else {
-      node.updateWith(config: _config, childrenInInversePaintOrder: children);
-    }
-    result.add(node);
-    // Sibling node needs to attach to the parent of an explicit node.
-    for (final SemanticsNode siblingNode in siblingNodes) {
-      // sibling nodes are in the same coordinate of the immediate explicit node.
-      // They need to share the same transform if they are going to attach to the
-      // parent of the immediate explicit node.
-      assert(siblingNode.transform == null);
-      siblingNode.transform = node.transform;
-      if (_tagsForChildren != null) {
-        siblingNode.tags ??= <SemanticsTag>{};
-        siblingNode.tags!.addAll(_tagsForChildren!);
+  void didUpdateParentData(_SemanticsParentData newParentData) {
+    if (parentData == newParentData) {
+      return;
+    }
+    // Parent data changes may result in node formation changes.
+    markNeedsBuild();
+    parentData = newParentData;
+    updateChildren();
+  }
+
+  void updateChildren() {
+    assert(parentData != null || isRoot, 'parent data can only be null for root rendering object');
+    configProvider.reset();
+    final Set<SemanticsTag>? tagsForChildren = _getTagsForChildren();
+    final bool explicitChildNodesForChildren =
+        isRoot ||
+        configProvider.effective.explicitChildNodes ||
+        // ParentData's explicitChildNode only
+        // propagate to children if this node doesn't
+        // contribute to semantics tree
+        (!contributesToSemanticsTree && (parentData?.explicitChildNodes ?? true));
+    final List<SemanticsConfiguration> childConfigurations = <SemanticsConfiguration>[];
+    final ChildSemanticsConfigurationsDelegate? childConfigurationsDelegate =
+        configProvider.effective.childConfigurationsDelegate;
+    final bool hasChildConfigurationsDelegate = childConfigurationsDelegate != null;
+    final Map<SemanticsConfiguration, _SemanticsFragment> configToFragment =
+        <SemanticsConfiguration, _SemanticsFragment>{};
+    final List<_SemanticsFragment> children = <_SemanticsFragment>[];
+    final bool blocksUserAction =
+        (parentData?.blocksUserActions ?? false) || configProvider.effective.isBlockingUserActions;
+    siblingMergeGroups.clear();
+    _childrenAndElevationAdjustments.clear();
+    mergeUp.clear();
+    for (final _RenderObjectSemantics childSemantics in _getNonBlockedChildren()) {
+      assert(!childSemantics.renderObject._needsLayout);
+      final _SemanticsParentData childParentData = _SemanticsParentData(
```

Comment:
are the childParentData the same for every child? Should one object be reused for all children?